packages/utils/typescript/lib/generators/schemas/index.js

```diff
@@ -0,0 +1,173 @@
+'use strict';
+
+const path = require('path');
+
+const ts = require('typescript');
+const { factory } = require('typescript');
+
+const fp = require('lodash/fp');
+const fse = require('fs-extra');
+const prettier = require('prettier');
+const chalk = require('chalk');
+const CLITable = require('cli-table3');
+
+const { generateImportDefinition } = require('./imports');
+const { generateSchemaDefinition } = require('./schema');
+const { generateGlobalDefinition } = require('./global');
+const {
+  getAllStrapiSchemas,
+  getSchemaInterfaceName,
+  getSchemaModelType,
+  getDefinitionAttributesCount,
+} = require('./utils');
+
+const DEFAULT_OUT_FILENAME = 'schemas.d.ts';
+
+/**
+ * Generate type definitions for Strapi schemas
+ *
+ * @param {object} options
+ * @param {Strapi} options.strapi
+ * @param {{ distDir: string; appDir: string; }} options.dirs
+ * @param {string} [options.outDir]
+ * @param {string} [options.file]
+ * @param {boolean} [options.debug]
+ */
+const generateSchemasDefinitions = async (options = {}) => {
+  const { strapi, outDir = process.cwd(), file = DEFAULT_OUT_FILENAME, debug = false } = options;
+
+  const schemas = getAllStrapiSchemas(strapi);
+
+  const schemasDefinitions = Object.values(schemas).map(schema => ({
+    schema,
+    definition: generateSchemaDefinition(schema),
+  }));
+
+  const allDefinitions = [
+    // Imports
+    generateImportDefinition(),
+
+    // Add a newline after the import statement
+    factory.createIdentifier('\n'),
+
+    // Schemas
+    ...schemasDefinitions.reduce(
+      (acc, def) => [
+        ...acc,
+        def.definition,
+        // Add a newline between each interface declaration
+        factory.createIdentifier('\n'),
+      ],
+      []
+    ),
+
+    // Global
+    generateGlobalDefinition(schemasDefinitions),
+  ];
+
+  const output = emitDefinitions(allDefinitions);
+  const formattedOutput = await format(output);
+
+  const definitionFilepath = await saveDefinitionToFileSystem(outDir, file, formattedOutput);
+
+  if (debug) {
+    logDebugInformation(schemasDefinitions, { filepath: definitionFilepath });
+  }
+};
+
+const emitDefinitions = definitions => {
+  const nodeArray = factory.createNodeArray(definitions);
+
+  const sourceFile = ts.createSourceFile(
```

Comment:
Just wondering if it is going to be using the user tsconfig somehow ?

Reply:
From what I've understood of the code it won't use it to generate the fake file, no.
Here's it's only building a compound node with all the declarations. You can then print it into human-readable text and write it in a file.

Reply:
I think we need to at least support the default configuration we're generating along with our TS projects. On top of that, we're following `strict: true` rules & thus don't have `any` types anywhere in the generated definitions.

IMO, if a user's custom configuration breaks the typings, it's gonna be mainly their responsibility to fix the conflicts, and in some scenarios, we can imagine changing the generation of the type to handle specific edge cases.

What's your opinion?

Reply:
Yeah let's see we might en dup having to read the conf and adapt our generators based on it if this happens